common/src/main/java/io/novafoundation/nova/common/validation/InputValidationMixin.kt

```diff
@@ -0,0 +1,26 @@
+package io.novafoundation.nova.common.validation
+
+import kotlinx.coroutines.flow.Flow
+
+interface InputValidationMixin {
+    val fieldValidationResult: Flow<FieldValidationResult>
+
+    interface Factory {
+        fun create(inputStream: Flow<String>, validator: FieldValidator)
+    }
+}
+
+class RealInputValidationMixinFactory(val validator: FieldValidator) {
+
+    fun create(inputStream: Flow<String>): InputValidationMixin {
+        return RealInputValidationMixin(inputStream, validator)
+    }
```

Comment:
I think it should be this way:

``` kotlin
RealInputValidationMixinFactory : InputValidationMixin.Factory {
   override fun create(inputStream: Flow<String>, validator: FieldValidator): InputValidationMixin {
        return RealInputValidationMixin(inputStream, validator)
    }
}
```

Othrewise we need to create a factory to create mixin, which ruins the purpose of the factory pattern